documentation/library_guide/parallel_api/pass_data_algorithms.rst

```diff
@@ -81,7 +81,19 @@ the buffer were created for the same queue. For example:
     return 0;
   }
 
-Alternatively, use ``std::vector`` with a USM allocator. For example:
+Alternatively, use ``std::vector`` with a USM allocator.
+Note: The ability to appropriately detect USM allocated ``std::vector::iterator`` depends
+on details of the C++ standard library implementation and what information about the
+allocator is included in the ``std::vector::iterator`` type definition. You can check
+if your C++ standard library implementation provides enough information by checking the
+value of ``oneapi::dpl::usm_allocated_vector_iterators_supported_v<ValueType>``. This will
```

Comment:
I would suggest the male another PR for that change, because introducing new piblic API stuff like `oneapi::dpl::usm_allocated_vector_iterators_supported_v` should be discussed and (approved) with oneDPL library architect.

Reply:
Separated out to #1476, but kept some of the documentation to describe some nuance to the support of USM allocators.